modules/gdscript/gdscript_parser.cpp

```diff
@@ -3418,8 +3426,12 @@ GDScriptParser::ExpressionNode *GDScriptParser::parse_call(ExpressionNode *p_pre
 				}
 				make_completion_context(COMPLETION_ATTRIBUTE_METHOD, call->callee);
 			} else {
-				// TODO: The analyzer can see if this is actually a Callable and give better error message.
-				push_error(R"*(Cannot call on an expression. Use ".call()" if it's a Callable.)*");
+				String type_name = call->callee->get_datatype().to_string();
+				if (type_name != "Variant" && type_name != "Callable") {
+					push_error(vformat(R"*(Type "%s" is not callable. Use ".call()" if it's a dynamic Callable.)*", type_name));
+				} else {
+					push_error(R"*(Cannot call on this expression. Use ".call()" if it's a Callable.)*");
+				}
```

Comment:
This TODO refers to improving error checking at the static analysis stage. At the parsing stage, data types are not yet resolved.

Reply:
I'll look into this if I can achieve this or not.

Reply:
I've added the changes to gdscript_analyzer.cpp which are hopefully what was requested from this TODO.